tests/e2e-playwright/tests/conftest.py

```diff
@@ -3,28 +3,192 @@
 # pylint: disable=unused-variable
 # pylint: disable=too-many-arguments
 # pylint: disable=too-many-statements
+# pylint: disable=no-name-in-module
+
+import os
+import random
+import re
+from collections.abc import Callable, Iterator
 
 import pytest
-from playwright.sync_api import BrowserContext, Page, APIRequestContext
-from typing import Iterator
+from faker import Faker
+from playwright.sync_api import APIRequestContext, BrowserContext, Page, WebSocket
+from pydantic import AnyUrl, TypeAdapter
+from pytest_simcore.playwright_utils import (
+    AutoRegisteredUser,
+    SocketIOEvent,
+    SocketIOProjectStateUpdatedWaiter,
+    decode_socketio_42_message,
+)
 
 
-@pytest.fixture
+def pytest_addoption(parser: pytest.Parser) -> None:
+    group = parser.getgroup(
+        "oSparc e2e options", description="oSPARC-e2e specific parameters"
+    )
+    group.addoption(
+        "--product-url",
+        action="store",
+        type=AnyUrl,
+        default=None,
+        help="URL pointing to the deployment to be tested",
+    )
+    group.addoption(
+        "--autoregister",
+        action="store_true",
+        default=False,
+        help="User name for logging into the deployment",
+    )
+    group.addoption(
+        "--user-name",
+        action="store",
+        type=str,
+        default=None,
+        help="User name for logging into the deployment",
+    )
+    group.addoption(
+        "--password",
+        action="store",
+        type=str,
+        default=None,
+        help="Password for logging into the deployment",
+    )
+    group.addoption(
+        "--product-billable",
+        action="store_true",
+        default=False,
+        help="Whether product is billable or not",
+    )
+    group.addoption(
+        "--service-test-id",
+        action="store",
+        type=str,
+        default=None,
+        help="Service test ID",
+    )
+    group.addoption(
+        "--service-key",
+        action="store",
+        type=str,
+        default=None,
+        help="Service Key",
+    )
+
+
+@pytest.fixture(autouse=True)
 def osparc_test_id_attribute(playwright):
     # Set a custom test id attribute
     playwright.selectors.set_test_id_attribute("osparc-test-id")
 
 
 @pytest.fixture
 def api_request_context(context: BrowserContext):
+    return context.request
+
+
+@pytest.fixture
+def product_url(request: pytest.FixtureRequest) -> AnyUrl:
+    if passed_product_url := request.config.getoption("--product-url"):
+        return TypeAdapter(AnyUrl).validate_python(passed_product_url)
+    return TypeAdapter(AnyUrl).validate_python(os.environ["PRODUCT_URL"])
+
+
+@pytest.fixture
+def user_name(request: pytest.FixtureRequest, auto_register: bool, faker: Faker) -> str:
+    if auto_register:
+        faker.seed_instance(random.randint(0, 10000000000))  # noqa: S311
```

Comment:
Are you sure you really want to make it random? I guess this is different from run to run, no? I understand the test should be independent of that choice, but I usually try to avoid any randomness in tests, unless it is strictly needed

Reply:
Currently the autoregister mode is mostly made for being used when you deploy locally or for later Github CI where it needs to be able to register to run automatically.
In effective deployments usually you cannot register anyway, so this would not be used there.

Reply:
nevertheless, I will add prefix in any case.